crates/ruff_linter/src/rules/isort/annotate.rs

```diff
@@ -116,6 +116,35 @@ pub(crate) fn annotate_imports<'a>(
                         })
                         .collect();
 
+                    // Capture trailing comments on the _last_ alias, as in:
+                    // ```python
+                    // from foo import (
+                    //     bar,
+                    //     # noqa
+                    // )
+                    // ```
+                    if let Some(last_alias) = aliases.last_mut() {
+                        while let Some(comment) =
+                            comments_iter.next_if(|comment| comment.start() < import.end())
+                        {
+                            last_alias.inline.push(comment);
+                        }
+                    }
+
+                    // Capture trailing comments, as in:
+                    // ```python
+                    // from foo import (
+                    //     bar,
+                    // )  # noqa
+                    // ```
+                    let mut after = vec![];
+                    let import_line_end = locator.line_end(import.end());
+                    while let Some(comment) = comments_iter.next_if(|comment| {
+                        import.end() <= comment.start() && comment.start() < import_line_end
```

Comment:
Shouldn't all remaining comments be guaranteed to start after (or at) `import.end`?

```suggestion
                    while let Some(comment) = comments_iter.next_if(|comment| {
                        comment.start() < import_line_end
```

Reply:
Good call.